javagen/src/main/java/com/azure/autorest/template/ModelTemplate.java

```diff
@@ -95,7 +95,7 @@ public final void write(ClientModel model, JavaFile javaFile) {
         javaFile.declareImport(imports);
 
         javaFile.javadocComment(settings.getMaximumJavadocCommentWidth(),
-            comment -> comment.description(model.getDescription()));
+            comment -> comment.description(model.getDescription(), JavaSettings.getInstance().isDataPlaneClient()));
```

Comment:
Thinking of a protected `boolean supportPartialUpdate()` method in the class.

This way, it could control all related features about partial-update, including this and `@Generated`.
One reason is that `<!-- start generated doc -->` without `@Generated` is probably useless (I assume we still check the existance of `@Generated` annotation for whether the class supports partial-update during processing).

Reply:
I think this is a good way to remove the generated javadoc wrapper if some classes do not want. I will update.